src/test/java/io/weaviate/integration/client/schema/ClusterSchemaTest.java

```diff
@@ -494,4 +448,40 @@ public void shouldBatchObjectsWithNestedProperties_NoSchema2() {
     schemaClass = utils.getClass(client2, className);
     utils.assertThatSchemaPropertiesHaveDataTypes(expectedProps, schemaClass);
   }
+
+  private void validateClassReplicationSettings(int replicationFactor, boolean asyncEnabled) {
+    // given
+    ReplicationConfig.ReplicationConfigBuilder replicationConfigBuilder = ReplicationConfig.builder().factor(replicationFactor);
+    if (asyncEnabled) {
+      replicationConfigBuilder = replicationConfigBuilder.asyncEnabled(true);
+    }
+    String className = "Band";
+    WeaviateClass clazz = WeaviateClass.builder()
+      .className(className)
+      .description("Band that plays and produces music")
+      .vectorIndexType("hnsw")
+      .vectorizer("text2vec-contextionary")
+      .replicationConfig(replicationConfigBuilder.build())
+      .build();
+
+    // when
+    Result<Boolean> createStatus = client1.schema().classCreator().withClass(clazz).run();
+    assertThat(createStatus).isNotNull()
+      .returns(false, Result::hasErrors)
+      .returns(true, Result::getResult);
+
+    // then
+    Result<WeaviateClass> classResult = client1.schema().classGetter().withClassName(className).run();
+    assertThat(classResult).isNotNull()
+      .returns(false, Result::hasErrors)
+      .extracting(Result::getResult).isNotNull()
+      .extracting(WeaviateClass::getReplicationConfig)
+      .isNotNull()
+      .satisfies(rc -> {
+        assertThat(rc.getFactor()).isEqualTo(replicationFactor);
+        if (asyncEnabled) {
+          assertThat(rc.getAsyncEnabled()).isEqualTo(true);
+        }
```

Comment:
what should be incoming value when `asyncEnabled == false`? test will not verify if it is not true

Reply:
I will change this to big `Boolean` and check if it's not null, then it means that we purposely have set it and then we want to check the outcome